devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+ 	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name:"thermostatFull", type:"thermostat", width:6, height:4) {
+			tileAttribute("device.temperature", key: "PRIMARY_CONTROL") {
+				attributeState("temp", label:'${currentValue}', unit:"dF", defaultState: true)
+			}
+			tileAttribute("device.temperature", key: "VALUE_CONTROL") {
+				attributeState("VALUE_UP", action: "tempUp")
+				attributeState("VALUE_DOWN", action: "tempDown")
+			}
+			tileAttribute("device.humidity", key: "SECONDARY_CONTROL") {
+				attributeState("humidity", label:'${currentValue}%', unit:"%", defaultState: true)
+			}
+			tileAttribute("device.thermostatOperatingState", key: "OPERATING_STATE") {
+				attributeState("idle", backgroundColor:"#00A0DC")
+				attributeState("heating", backgroundColor:"#e86d13")
+				attributeState("cooling", backgroundColor:"#00A0DC")
+			}
+			tileAttribute("device.thermostatMode", key: "THERMOSTAT_MODE") {
+				attributeState("off", label:'${name}')
+				attributeState("heat", label:'${name}')
+				attributeState("cool", label:'${name}')
+				attributeState("auto", label:'${name}')
+			}
+			tileAttribute("device.heatingSetpoint", key: "HEATING_SETPOINT") {
+				attributeState("heatingSetpoint", label:'${currentValue}', unit:"dF", defaultState: true)
+			}
+			tileAttribute("device.coolingSetpoint", key: "COOLING_SETPOINT") {
+				attributeState("coolingSetpoint", label:'${currentValue}', unit:"dF", defaultState: true)
+			}
+		}
+
+		valueTile("temperature", "device.temperature", width: 2, height: 2) {
+			state("temperature", label:'${currentValue}', unit:"dF",
+				backgroundColors:[
+							// Celsius
+							[value: 0, color: "#153591"],
+							[value: 7, color: "#1e9cbb"],
+							[value: 15, color: "#90d2a7"],
+							[value: 23, color: "#44b621"],
+							[value: 28, color: "#f1d801"],
+							[value: 35, color: "#d04e00"],
+							[value: 37, color: "#bc2323"],
+							// Fahrenheit
+							[value: 40, color: "#153591"],
+							[value: 44, color: "#1e9cbb"],
+							[value: 59, color: "#90d2a7"],
+							[value: 74, color: "#44b621"],
+							[value: 84, color: "#f1d801"],
+							[value: 95, color: "#d04e00"],
+							[value: 96, color: "#bc2323"]
+				]
+			)
+		}
+		standardTile("tempDown", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "tempDown", label:'down', action:"tempDown", defaultState: true , icon:"st.thermostat.thermostat-down"
+		}
+		standardTile("tempUp", "device.temperature", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
+			state "tempUp", label:'up', action:"tempUp", defaultState: true , icon:"st.thermostat.thermostat-up"
+		}
+
+		valueTile("heatingSetpoint", "device.heatingSetpoint", width: 2, height: 2, inactiveLabel: false, decoration: "flat") {
```

Comment:
heatingSetpoint and coolingSetpoint, should both be set by the tileAttribute thermostatSetpoint multiAttributeTile, if there's a need to keep the heatingSetpoint and coolingSetpoint in separate tiles then the multiAttributeTile thermostatSetpoint should be eliminated.

Reply:
Fair point.  Some other DTHs on the platform also have multiAttribute and separate cool/heat tiles (and some DTHs do not).  Having the separate tiles (although clutter), is useful for property managers.

Reply:
Which DTH's are those? the one I've worked with have on or the other, mostly separate heating/cooling setpoint due to how thermostats are handling them. Also, those thermostats are using the generic version of the multiAttributeTile due to issues with the thermostat version that seems to have been fixed mostly now.

Reply:
removed all up/down tiles other than the main multiAttributeTile.